service/src/main/java/bio/terra/workspace/service/workspace/flight/cloud/aws/DeleteWorkspaceApplicationSecurityGroupsStep.java

```diff
@@ -0,0 +1,88 @@
+package bio.terra.workspace.service.workspace.flight.cloud.aws;
+
+import bio.terra.cloudres.aws.ec2.EC2SecurityGroupCow;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.workspace.AwsCloudContextService;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.ec2.model.Ec2Exception;
+
+public class DeleteWorkspaceApplicationSecurityGroupsStep implements Step {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(DeleteWorkspaceApplicationSecurityGroupsStep.class);
+  private final CrlService crlService;
+  private final SamService samService;
+  private final AwsCloudContextService awsCloudContextService;
+  private final UUID workspaceUuid;
+
+  public DeleteWorkspaceApplicationSecurityGroupsStep(
+      CrlService crlService,
+      AwsCloudContextService awsCloudContextService,
+      SamService samService,
+      UUID workspaceUuid) {
+    this.crlService = crlService;
+    this.samService = samService;
+    this.awsCloudContextService = awsCloudContextService;
+    this.workspaceUuid = workspaceUuid;
+  }
+
+  private void deleteSecurityGroup(
```

Comment:
share this method with the undo step above?

Reply:
Thanks, added a utility method to reuse and clean up the call site a bit.